### ISSUE TITLE
[flang][coarray] Unexpected error: 'fir.store' op store value type must match memory reference type

### ISSUE THREAD

**Issue by sscalpone** [llvm/llvm-project#193131]
```
% cat r.f90
```
```
 class(*),allocatable :: a[:]
 allocate(real::a[1:*])
 end
```
```
% flang -c -fcoarray r.f90
warning: Support for multi image Fortran features is still experimental and in development.
error: loc("r.f90":2:2): 'fir.store' op store value type must match memory reference type
error: verification of lowering to FIR failed
```

**Comment by JDPailleux** [comment]
Hi @sscalpone , I got it, I'll take care of it ;)
